### DISCUSSION TITLE
How to detect prime number using this package?

### DISCUSSION THREAD

**Discussion by lamualfa** [markrogoyski/math-php#479] [Q&A] [answered]
Is there any function to do this?

```php
use MathPHP\NumberTheory\Integer;

Integer::isPrime(5);
Integer::isPrime(2910);
Integer::isPrime(2832981);
```

**Comment by Beakerboy** [comment]
```php
count(Integer::primeFactorization(5)) == 1         // True
count(Integer::primeFactorization(2910)) == 1      // False
count(Integer::primeFactorization(2832981)) == 1   // ??
```

**Comment by markrogoyski** [comment, marked as answer]
I've implemented `NumberTheory\Integer::isPrime($n)` in the latest version [v2.10.0](https://github.com/markrogoyski/math-php/releases/tag/v2.10.0). Thank you @lamualfa for the suggestion, and @Beakerboy for the temporary workaround suggestion.

**Reply by lamualfa** [reply]
![image](https://github.com/markrogoyski/math-php/assets/39755201/7c4f9694-21df-48ea-bd28-10571c7c7fa0)

https://en.wikipedia.org/wiki/Primality_test
